indexer/app/lib/periodic_indexer.rb

```diff
@@ -278,72 +296,127 @@ def run_index_round
 
       checkpoints = []
 
-      did_something = false
```

Comment:
@cfitz - why did this line move into the block below? It will raise and undefined exception if nothing happens in any of the record types...